scalding-core/src/main/scala/com/twitter/scalding/serialization/JavaStreamEnrichments.scala

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2015 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.twitter.scalding.serialization
+
+import java.io.{ ByteArrayInputStream, ByteArrayOutputStream, InputStream, OutputStream, EOFException }
+
+object JavaStreamEnrichments {
+  def eof: Nothing = throw new EOFException()
+
+  implicit class RichByteArrayOutputStream(val baos: ByteArrayOutputStream) extends AnyVal {
+    def toInputStream: ByteArrayInputStream = new ByteArrayInputStream(baos.toByteArray)
+  }
+
+  /**
+   * This has a lot of methods from DataInputStream without
+   * having to allocate to get them
+   * This code is similar to those algorithms
+   */
+  implicit class RichInputStream(val s: InputStream) extends AnyVal {
+    /**
+     * If s supports marking, we mark it. Otherwise we read the needed
+     * bytes out into a ByteArrayStream and return that.
+     * This is intended for the case where you need possibly
+     * read size bytes but may stop early, then skip this exact
+     * number of bytes.
+     * Intended use is:
+     * {code}
+     * val size = 100
+     * val marked = s.markOrBuffer(size)
+     * val y = fn(marked)
+     * marked.reset
+     * marked.skipFully(size)
+     * {/code}
+     */
+    def markOrBuffer(size: Int): InputStream = {
+      val ms = if (s.markSupported) s else {
+        val buf = new Array[Byte](size)
+        s.readFully(buf)
+        new ByteArrayInputStream(buf)
+      }
+      // Make sure we can reset after we read this many bytes
+      ms.mark(size)
+      ms
+    }
+
+    def readBoolean: Boolean = (readUnsignedByte != 0)
+
+    /**
+     * Like read, but throws eof on error
+     */
+    def readByte: Byte = readUnsignedByte.toByte
+
+    def readUnsignedByte: Int = {
+      val c1 = s.read
```

Comment:
could you put a type on c1? it looks like its int? So when you really read a byte from the underlying stream it gives you an unsigned byte in an int?